### PR TITLE
Remove session

### PR DESCRIPTION
**Pull request by andreamazza89**

### CODE REVIEW

File: lib/tic_tac_toe/web/app.ex [hunk 1]
```diff
@@ -3,7 +3,7 @@ defmodule TicTacToe.Web.App do
 
   def start(_type, _args) do
     port = Application.get_env(:elixir_web_ttt, :port)
-    IO.puts "Server running on port #{port}. Visit /tictactoe/options" <>
+    IO.puts "Server running on port #{port}. Visit /ttt/options" <>
```
Comment (c-j-j) [inline lib/tic_tac_toe/web/app.ex]: Can't you set the root path to redirect to /ttt/options so the user doesn't have to?

File: lib/tic_tac_toe/web/game_state_stringifier.ex [hunk 2]
```diff
@@ -0,0 +1,33 @@
+defmodule TicTacToe.Web.GameStateStringifier do
```
Comment (c-j-j) [inline lib/tic_tac_toe/web/game_state_stringifier.ex]: Not sure on the naming here, and using "Stringifier". Maybe "serialize"/"deserialize" would be more appropriate?

File: lib/tic_tac_toe/web/router.ex [hunk 3]
```diff
@@ -1,62 +1,54 @@
...
-    response_body = View.render_game(game_state)
+  get ("ttt/play/:stringy_game_state") do
+    game_state = GameStateStringifier.parse(stringy_game_state)
+    response_body = View.render_game(game_state, stringy_game_state)
     conn |> put_resp_content_type("html") |> resp(200, response_body)
   end
 
-  get ("/tictactoe/computer_move") do
-    conn |> GameSessionPlug.make_next_move() |> redirect_to("/tictactoe/play")
+  get ("/ttt/computer_move/:stringy_game_state") do
```
Comment (c-j-j) [inline lib/tic_tac_toe/web/router.ex]: The query parameter looks a little verbose;

`http://localhost:4000/ttt/play/%7B%22__struct__%22:%22Elixir.Game%22,%22board%22:%7B%22__struct__%22:%22Elixir.Board%22,%22cells%22:[%22x%22,%22empty%22,%22empty%22,%22empty%22,%22o%22,%22empty%22,%22empty%22,%22empty%22,%22empty%22]%7D,%22players%22:[%7B%22__struct__%22:%22Elixir.Player.Human%22,%22io%22:[%22Elixir.UI.Console%22,%22stdio%22],%22mark%22:%22x%22%7D,%7B%22__struct__%22:%22Elixir.Player.MiniMax%22,%22mark%22:%22o%22%7D]%7D`

The query parameters are user facing, so I'd try and make them more "user friendly" if you can call it that, i.e. /play-game?game-type=hvc&board=EXOXXOOXX

File: lib/tic_tac_toe/web/router.ex [hunk 4]
```diff
@@ -1,62 +1,54 @@
...
     conn |> put_resp_content_type("html") |> resp(200, response_body)
   end
 
-  get ("/tictactoe/play") do
-    game_state = conn |> GameSessionPlug.get_game_state()
-    response_body = View.render_game(game_state)
+  get ("ttt/play/:stringy_game_state") do
+    game_state = GameStateStringifier.parse(stringy_game_state)
+    response_body = View.render_game(game_state, stringy_game_state)
     conn |> put_resp_content_type("html") |> resp(200, response_body)
```
Comment (c-j-j) [inline lib/tic_tac_toe/web/router.ex:23]: I like that you've extracted most of the logic out of this router. Keeps things nice and readable, and ensures this is the boundary to your application and the rest of the logic is nested below.

File: lib/tic_tac_toe/web/game_state_stringifier.ex (hunk 2 above)
Comment (c-j-j) [reply lib/tic_tac_toe/web/game_state_stringifier.ex]: Or Maybe `GameStatePresenter`, with `to_json` and `from_json`?

File: lib/tic_tac_toe/web/router.ex [hunk 5]
```diff
@@ -1,62 +1,54 @@
...
+  get ("/ttt/computer_move/:stringy_game_state") do
+    old_game_state = GameStateStringifier.parse(stringy_game_state)
+    new_game_state = Game.make_next_move(old_game_state)
+    stringy_new_game_state = GameStateStringifier.stringify(new_game_state)
+    conn |> redirect_to("/ttt/play/#{stringy_new_game_state}")
   end
 
-  post ("/tictactoe/moves/:move") do
-    conn |> GameSessionPlug.update_game_state_with_move(move) |> redirect_to("/tictactoe/play")
+  post ("/ttt/moves/:move/:stringy_game_state") do
```
Comment (c-j-j) [inline lib/tic_tac_toe/web/router.ex]: I'm surprised you still need a `post` route, if the backend is stateless?

File: templates/stringy_game.eex [hunk 6]
```diff
@@ -24,7 +24,7 @@
                   <%= cell_presenter.mark(cell) %>
                 </div>
               <%= else %>
-                <form action="/tictactoe/moves/<%= cell_presenter.index(cell) %>" method="post" class="move-button">
+                <form action='/ttt/moves/<%= cell_presenter.index(cell) %>/<%= stringy_game_state %>' method="post" class="move-button">
```
Comment (c-j-j) [inline templates/stringy_game.eex]: not sure on the `stringy_game_state` variable name - feels more verbose than it needs to be. I somehow feel just having `game_state` would be more clearer? I can see why you would name it `stringy_` to differentiate between it and the actual `game_state`, but I think in the view it's a safe assumption that the variables in the view are modified.

File: lib/tic_tac_toe/web/app.ex (hunk 1 above)
Comment (andreamazza89) [reply lib/tic_tac_toe/web/app.ex]: Yes, I shall 👍

Comment (c-j-j) [reply lib/tic_tac_toe/web/app.ex]: Or have the root path configured to show the options page.

File: lib/tic_tac_toe/web/game_state_stringifier.ex (hunk 2 above)
Comment (andreamazza89) [reply lib/tic_tac_toe/web/game_state_stringifier.ex]: I agree, stringy string mc string just does not sound right.

File: lib/tic_tac_toe/web/router.ex (hunk 4 above)
Comment (andreamazza89) [reply lib/tic_tac_toe/web/router.ex:23]: Thanks!

File: lib/tic_tac_toe/web/router.ex (hunk 5 above)
Comment (andreamazza89) [reply lib/tic_tac_toe/web/router.ex]: mm interesting, in my head, the reason for using POST is  that the client is sending information to the server, which then updates the state and redirects to the main playing page (with the new state embedded)...though I guess I am going against the convention of using POST only for a  request where the information changes the state of the server? In which case should I change this route to a GET or is it a more fundamental problem?

Comment (c-j-j) [reply lib/tic_tac_toe/web/router.ex]: I would only use `post` only when the backend is actually changed. i.e. the database was modified etc. If you're only returning updated state, I think a `get` would be more idiomatic.

Comment (andreamazza89) [reply lib/tic_tac_toe/web/router.ex]: perfect, thank you :)

File: lib/tic_tac_toe/web/router.ex (hunk 3 above)
Comment (andreamazza89) [reply lib/tic_tac_toe/web/router.ex]: Done, it now looks prettier, like so:

`GET /ttt/play/Board=x,x,o,o,o,x,x,o,x-Players=Human,o,Human,x`

Comment (c-j-j) [reply lib/tic_tac_toe/web/router.ex]: Do those commas not need URL Encoding?

**Comment by andreamazza89** [comment]
Thank you so much for your feedback! :bowtie:

**Pull request merged**